frontend/pkg/frontend/ocm.go

```diff
@@ -243,22 +242,15 @@ func ConvertCStoHCPOpenShiftCluster(resourceID *azcorearm.ResourceID, cluster *a
 		},
 	}
 
-	// Temporarily set KeyManagementMode to its default value.
-	hcpcluster.Properties.Etcd = api.EtcdProfile{
-		DataEncryption: api.EtcdDataEncryptionProfile{
-			KeyManagementMode: api.EtcdDataEncryptionKeyManagementModeTypePlatformManaged,
-		},
+	// Only set etcd encryption settings if they exist in the cluster service response
+	if cluster.Azure().EtcdEncryption() != nil && cluster.Azure().EtcdEncryption().DataEncryption() != nil {
+		hcpcluster.Properties.Etcd = api.EtcdProfile{
+			DataEncryption: api.EtcdDataEncryptionProfile{
+				CustomerManaged:   convertCustomerManagedEncryptionCSToRP(cluster.Azure().EtcdEncryption().DataEncryption()),
+				KeyManagementMode: convertKeyManagementModeTypeCSToRP(cluster.Azure().EtcdEncryption().DataEncryption().KeyManagementMode()),
+			},
+		}
```

Comment:
Let's introduce a `dataEncryption` variable here to eliminate some repetition.
```suggestion
	// Only set etcd encryption settings if they exist in the cluster service response
	if cluster.Azure().EtcdEncryption() != nil {
		dataEncryption := cluster.Azure.EtcdEncryption().DataEncryption()
		if dataEncryption != nil {
				hcpcluster.Properties.Etcd = api.EtcdProfile{
					DataEncryption: api.EtcdDataEncryptionProfile{
						CustomerManaged:   convertCustomerManagedEncryptionCSToRP(dataEncryption),
						KeyManagementMode: convertKeyManagementModeTypeCSToRP(dataEncryption.KeyManagementMode()),
					},
			},
		}
```